README.md

```diff
@@ -1,14 +1,50 @@
 
-# sessionOffload
+# OpenOffload
 
 gRPC API to offload TCP and UDP packet processing from an application to a hardware switch
 
-# Requisites
+# Creating Language Bindings
+
+There are three language bindings implemented in the language Makefile.
+
+- Python
+- Golang
+- C++
+
+To generate the language bindings for a specific language the appropriate language and related gRPC tools need to be installed. It is the responsibility of individual implementer to create there client and server code. 
+
+## Generating code
+
+### Python
+
+```bash
+$ cd openoffload
+$ make all-py
+```
+
+### Golang
+
+```bash
+$ cd openoffload
+$ make all-go
+```
+### C++
```

Comment:
should we have a reference to apt get install protobuf-compiler

Reply:
hmm i guess that isnt quite sufficient

(base) root@njcdtl01bf1936:/home2/bf1936/WB/sessionOffload.v1alpha2/openoffload# make all-cpp
protoc -I ../protos --cpp_out=cpp openoffload.proto
openoffload.proto: No such file or directory
Makefile:32: recipe for target 'openoffload.pb.cc' failed
make: *** [openoffload.pb.cc] Error 1

(base) root@njcdtl01bf1936:/home2/bf1936/WB/sessionOffload.v1alpha2/openoffload# protoc -I ../protos --cpp_out=cpp ../protos/openoffload.proto
openoffload.proto:18:10: Unrecognized syntax identifier "proto3".  This parser only recognizes "proto2".

Reply:
need proto-c 3.0 or greater - default install is 2.6.1 which doesnt support proto3

Reply:
Yes I went through the instructions on the gRPC site - it was a little more complicated and is platform dependent.

Reply:
I think we should not generate the code and put the generated code files in .gitignore. For different architectures x86, ARM etc the code will be different so we should either do all or none.